WooCommerce/Classes/ViewRelated/Orders/Order Details/Order Summary Section/SummaryTableViewCellViewModel.swift

```diff
@@ -0,0 +1,73 @@
+import Foundation
+import Yosemite
+
+struct SummaryTableViewCellViewModel {
+    struct OrderStatusPresentation {
+        let style: OrderStatusEnum
+        let title: String
+    }
+
+    private let billingAddress: Address?
+    private let dateCreated: Date
+    private let salesChannel: String?
+
+    let presentation: OrderStatusPresentation
+
+    private let calendar: Calendar
+
+    init(order: Order,
+         status: OrderStatus?,
+         calendar: Calendar = .current) {
+
+        billingAddress = order.billingAddress
+        dateCreated = order.dateCreated
+        salesChannel = order.createdVia
+
+        presentation = OrderStatusPresentation(
+            style: status?.status ?? order.status,
+            title: status?.name ?? order.status.rawValue
+        )
+
+        self.calendar = calendar
+    }
+
+    /// The full name from the billing address
+    ///
+    var billedPersonName: String {
+        if let fullName = billingAddress?.fullName, fullName.isNotEmpty {
+            return fullName
+        } else {
+            return Localization.guestName
+        }
+    }
+
+    /// The date, time, and the order number concatenated together. Example, “Jan 22, 2018, 11:23 AM”.
+    ///
+    var subtitle: String {
+        let formatter = DateFormatter.dateAndTimeFormatter
+        formatter.timeZone = .siteTimezone
+        return formatter.string(from: dateCreated)
+    }
+
+    /// Textual representation of the sales channel
+    ///
+    var formattedSalesChannel: String? {
+        guard let salesChannel = salesChannel else {
+            return nil
+        }
+        switch salesChannel {
+        case "pos-rest-api":
+            return "POS"
+        default:
+            return nil
+        }
+    }
```

Comment:
This is a code duplication with the order list badge that we can resolve moving the logic to Yosemite, I'll investigate this separately tomorrow as there's some build errors when trying to extend the Order, otherwise we can use some sort of simple object to share logic from Yosemite without needing to extend anything. Ref: p1751433227323889-slack-CGPNUU63E